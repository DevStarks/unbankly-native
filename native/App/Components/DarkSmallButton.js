import React, { Component } from 'react'
import styles from './Styles/DarkSmallButtonStyles'
import ExamplesRegistry from '../Services/ExamplesRegistry'
import SmallButton from './SmallButton'

ExamplesRegistry.addComponentExample('DarkSmallButton', () =>
  <DarkSmallButton
    text='small'
    onPress={() => window.alert('Your button was pressed')}
  />
)

class DarkSmallButton extends Component {
  // see SmallButton for prop info
  render () {
    return (
      <SmallButton
        {...this.props}
        textStyle={Object.assign({}, styles.buttonText, this.props.textStyle)}
        style={[styles.button, this.props.style]} />
    )
  }
}

export default DarkSmallButton
