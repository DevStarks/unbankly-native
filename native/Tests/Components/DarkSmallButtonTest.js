import 'react-native'
import React from 'react'
import DarkSmallButton from '../../App/Components/DarkSmallButton'
import renderer from 'react-test-renderer'

test('DarkSmallButton component renders correctly', () => {
  const tree = renderer.create(<DarkSmallButton onPress={() => {}} text='hi' />).toJSON()
  expect(tree).toMatchSnapshot()
})
