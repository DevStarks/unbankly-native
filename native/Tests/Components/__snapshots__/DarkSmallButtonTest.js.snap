// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DarkSmallButton component renders correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#637381",
      "opacity": 1,
      "paddingVertical": 10,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#637381",
          "fontFamily": "Oswald-Bold",
          "fontSize": 20,
          "textAlign": "center",
        },
        Object {
          "color": "#F2EFEB",
        },
      ]
    }
  >
    hi
  </Text>
</View>
`;
